src/freenas/usr/lib/netdata/python.d/cputemp.chart.py

```diff
@@ -68,11 +68,13 @@ def get_data(self):
                 cpu_data[chip_name] = cpu_d
         except sensors.SensorsError as error:
             self.error(error)
-            return None
 
-        data = {}
-        for core, temp in cpu_temperatures(cpu_data).items():
-            data[str(core)] = temp
+        try:
+            data = {}
+            for core, temp in cpu_temperatures(cpu_data).items():
+                data[str(core)] = temp
```

Comment:
If we have 8 cores and get an exception when processing, for example, 3rd, `data` will have only two values while it should have 8

Reply:
@themylogin error if occurs would happen in `cpu_temperatures` implementation - i don't see it happening during iteration on some specific core..should we handle that too separately?

Reply:
In that case only `cpu_temperatures` should be inside the exception clause... broad exception clause especially.